plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts

```diff
@@ -207,32 +209,64 @@ export class ArgoService implements ArgoServiceApi {
     return data;
   }
 
-  async getArgoToken(appConfig: {
-    url: string;
-    username?: string;
-    password?: string;
-  }): Promise<string> {
-    const { url, username, password } = appConfig;
+  async getArgoToken(
+    argoInstanceConfig: InstanceConfig,
+    azureCredentials?: AzureConfig,
+  ): Promise<string> {
+    const { url, username, password, token } = argoInstanceConfig;
 
-    const options: RequestInit = {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({
-        username: username || this.username,
-        password: password || this.password,
-      }),
-    };
-    const resp = await fetch(`${url}/api/v1/session`, options);
-    if (!resp.ok) {
-      this.logger.error(`failed to get argo token: ${url}`);
+    if (token) return token;
+
+    if ((username && password) || (this.username && this.password)) {
+      const resp = await fetch(`${url}/api/v1/session`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          username: username || this.username,
+          password: password || this.password,
+        }),
+      });
+      if (!resp.ok) {
+        this.logger.error(`failed to get argo token: ${url}`);
+      }
+      if (resp.status === 401) {
+        throw new Error(`Getting unauthorized for Argo CD instance ${url}`);
+      }
+      const data = await resp.json();
+      return data.token;
     }
-    if (resp.status === 401) {
-      throw new Error(`Getting unauthorized for Argo CD instance ${url}`);
+
+    if (azureCredentials) {
+      const resp = await fetch(
+        `${azureCredentials.loginUrl}/${azureCredentials.tenantId}/oauth2/v2.0/token`,
+        {
+          method: 'POST',
+          body: qs.stringify({
```

Comment:
I'm not familiar with `qs`. If you just need query strings encoded, couldn't you just use `URLSearchParams`? Then you wouldn't need the library at all.

```suggestion
          body: new URLSearchParams({
```

Reply:
Hey! Thanks for the suggestion. We learned something new. We went ahead and changed it and we are removing the package.